packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -342,40 +326,18 @@ export function TransferPanel() {
     if (!selectedToken) {
       return
     }
-    if (!walletAddress) {
-      return
+    if (!sourceChainSigner) {
+      throw Error(signerUndefinedError)
     }
-    const signer = isDepositMode ? parentSigner : childSigner
-    if (!signer) {
-      throw 'Signer is undefined'
+    const signer = sourceChainSigner
+
+    if (!(await isTransferAllowed(true))) {
```

Comment:
this boolean prop is not very clear, but maybe it will be gone if we refactor the method, I don't think we need props here at all, this could probably be done with `useMemo`